### PR TITLE
[CPU] Add PagedAttention support

### PR DESCRIPTION
**Pull request by luo-cheng2021**
### Details:
 - *Support PagedAttention support, depends on:*
    - openvino_contrib: https://github.com/openvinotoolkit/openvino_contrib/pull/867
    - vLLM: https://github.com/ilya-lavrenov/vllm/pull/4
 - *TODO*
    - Models with alibi feature
   
### Tickets:
 - *[134329](https://jira.devtools.intel.com/browse/CVS-134329)*
 - *[134327](https://jira.devtools.intel.com/browse/CVS-134327)*

### CODE REVIEW

**Review by usstq — COMMENTED** [review]
Can we split page attention code into a separate CPU node? although `PageAttention`  share some underlying kernels with `ScaledDotProductAttention` , but the definition of the Node deserve a separate Node type, just for more maintainability.

**Review by usstq — APPROVED** [review]
approve and looking forward to refactor later, great works!

File: src/plugins/intel_cpu/src/cpu_types.cpp [hunk 1]
```diff
@@ -217,6 +217,7 @@ static const TypeToNameMap& get_type_to_name_tbl() {
         {"Ngram", Type::Ngram},
         {"ScaledDotProductAttention", Type::ScaledDotProductAttention},
         {"ScaledDotProductAttentionWithKVCache", Type::ScaledDotProductAttention},
+        {"PagedAttentionExtension", Type::ScaledDotProductAttention},
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/cpu_types.cpp:220]: Will do the refactor in the next PR, ticket: [136170](https://jira.devtools.intel.com/browse/CVS-136170)

File: src/plugins/intel_cpu/src/graph.cpp [hunk 2]
```diff
@@ -1680,6 +1680,10 @@ void Graph::EnforceInferencePrecision() {
                 if (node->getOriginalInputPrecisionAtPort(inPort) != ov::element::f32)
                     return true;
 
+                // kvcache of PagedAttention should be written directly
+                if (node->getType() == Type::ScaledDotProductAttention && node->getOriginalInputsNumber() == 13 &&
+                    (inPort == 3 || inPort == 4))
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/graph.cpp:1685]: Will do the refactor in the next PR, ticket: [136170](https://jira.devtools.intel.com/browse/CVS-136170)

[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/graph.cpp:1685]: > another tricky thing is, there seems to be another ways to keep original precision, which is to reject/ignore `getOriginalInputPrecision()` inside `initSupportedPrimitiveDescriptors()`, just report `fp32` in PortConfigurator description

In [line](https://github.com/openvinotoolkit/openvino/blob/67803b691cbabf00488faa4ebe50ceb745356779/src/plugins/intel_cpu/src/graph.cpp#L1720) the original precision will be changed, so `getOriginalInputPrecision()` will not work.

**Review by ilya-lavrenov — CHANGES_REQUESTED** [review]
Double checked - cache_dtype=float16 does not work and generates invalid output

**Comment by luo-cheng2021** [comment]
> Double checked - cache_dtype=float16 does not work and generates invalid output

Could you please try again using the latest version?

**Review by ilya-lavrenov — APPROVED** [review]
Passed with fp16 KV cache

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**